packages/rich-list-api-client/src/RichListApiClient.ts

```diff
@@ -0,0 +1,34 @@
+import 'url-search-params-polyfill'
+import { OceanApiClientOptions, OceanApiClient, ApiPagedResponse } from '@defichain/ocean-api-client'
+
+/**
+ * RichListApiClient configurable options
+ */
+export interface RichListApiClientOptions extends OceanApiClientOptions {}
+
+/**
+ * RichListApiClient
+ */
+export class RichListApiClient extends OceanApiClient {
+  constructor (
+    protected readonly options: RichListApiClientOptions
+  ) {
+    super({
+      url: 'https://rich-list.defichain.com', // TBD
+      timeout: 60000,
+      version: 'v0',
+      network: 'mainnet',
+      ...options
+    })
+    this.options.url = this.options.url?.replace(/\/$/, '')
+  }
+
+  async get (tokenId: string): Promise<ApiPagedResponse<RichListItem>> {
+    return await this.requestData('GET', `rich-list/${tokenId}`)
+  }
+}
+
+export interface RichListItem {
```

Comment:
In #1084, you are calling it `AddressBalance` and here you are using `RichListItem`, I understand that this is from the client perspective and thus `RichListItem` does makes a bit more sense. Is this intended?

Reply:
Yes, it is intentional.

In RichListCore, it hold ALL addresses with their known balance.
From client perspective, it retrieve only 1000 rows with highest balance.